### PR TITLE
test/mpi: remove arbitrary maxbuf limit

### PR DESCRIPTION
**Pull request by gcongiu**
## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by gcongiu** [comment]
test:mpich/warnings

File: test/mpi/util/mtest_common.c [hunk 1]
```diff
@@ -13,23 +13,11 @@
...
-    if (envval) {
-        max_size = atol(envval);
-        if (max_size < 100000) {
-            fprintf(stderr, "MPITEST_MAXBUFFER (%s) is too small, we need at least 100MB\n",
-                    envval);
-            exit(1);
-        }
-    }
-    return max_size;
+    return (1 << 30);
```
Comment (pavanbalaji) [inline test/mpi/util/mtest_common.c:20]: Why are you not allowing the user to override this?

**Comment by gcongiu** [comment]
test:mpich/warnings
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by gcongiu** [comment]
test:mpich/ch3/tcp

**Pull request merged**